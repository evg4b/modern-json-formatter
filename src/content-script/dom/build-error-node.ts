import { resource } from '@core/browser';
import { createElement } from '@core/dom';

export const buildErrorNode = (header: string, ...lines: string[]) => {
  const wrapper = document.createElement('div');
  wrapper.className = 'error';
  const image = document.createElement('img');

  image.src = resource('invalid.svg');
  image.alt = 'Error';
  const message = document.createElement('div');
  message.className = 'message';
  message.append(
    createElement({ element: 'span', content: header }), // 'Invalid json file.' }),
    ...lines.map(line => createElement({ element: 'span', content: line })),
    // element({ content: 'Please check the file and try again.' }),
  );
  wrapper.append(image, message);

  return wrapper;
};
