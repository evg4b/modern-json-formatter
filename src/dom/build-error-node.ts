import { element } from './helpres';

export const buildErrorNode = (header: string, ...lines: string[]) => {
  const wrapper = document.createElement('div');
  wrapper.className = 'error';
  const image = document.createElement('img');
  image.src = chrome.runtime.getURL('invalid.svg');
  image.alt = 'Error';
  const message = document.createElement('div');
  message.className = 'message';
  message.append(
    element({ content: header }), // 'Invalid json file.' }),
    ...lines.map(line => element({ content: line })),
    // element({ content: 'Please check the file and try again.' }),
  );
  wrapper.append(
    image,
    message,
  );

  return wrapper;
};
