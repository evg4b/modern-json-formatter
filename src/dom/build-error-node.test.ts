import { buildErrorNode } from './build-error-node';

describe('error-node', () => {
  beforeEach(() => {
    Object.assign(global, {
      chrome: {
        runtime: {
          getURL: jest.fn((a) => a),
        },
      },
    });
  });

  test('should render error node', () => {
    const node = buildErrorNode();

    expect(node).toMatchSnapshot();
  });
});
