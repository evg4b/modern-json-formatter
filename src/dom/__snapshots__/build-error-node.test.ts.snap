// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error-node should render error node 1`] = `
<div
  class="error"
>
  <img
    alt="Error"
    src="invalid.svg"
  />
  <div
    class="message"
  >
    <span>
      Invalid json file.
    </span>
    <span>
      Please check the file and try again.
    </span>
  </div>
</div>
`;
